src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -514,35 +508,33 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 
 	//#region openEditor()
 
-	openEditor(editor: IEditorInput, options?: IEditorOptions, group?: OpenInEditorGroup): Promise<IEditorPane | undefined>;
-	openEditor(editor: IResourceEditorInput, group?: OpenInEditorGroup): Promise<IEditorPane | undefined>;
-	openEditor(editor: ITextResourceEditorInput | IUntitledTextResourceEditorInput, group?: OpenInEditorGroup): Promise<ITextEditorPane | undefined>;
-	openEditor(editor: IResourceDiffEditorInput, group?: OpenInEditorGroup): Promise<ITextDiffEditorPane | undefined>;
-	async openEditor(editor: IEditorInput | IUntypedEditorInput, optionsOrPreferredGroup?: IEditorOptions | OpenInEditorGroup, preferredGroup?: OpenInEditorGroup): Promise<IEditorPane | undefined> {
+	openEditor(editor: IEditorInput, options?: IEditorOptions, group?: PreferredGroup): Promise<IEditorPane | undefined>;
+	openEditor(editor: IResourceEditorInput, group?: PreferredGroup): Promise<IEditorPane | undefined>;
+	openEditor(editor: ITextResourceEditorInput | IUntitledTextResourceEditorInput, group?: PreferredGroup): Promise<ITextEditorPane | undefined>;
+	openEditor(editor: IResourceDiffEditorInput, group?: PreferredGroup): Promise<ITextDiffEditorPane | undefined>;
+	async openEditor(editor: IEditorInput | IUntypedEditorInput, optionsOrPreferredGroup?: IEditorOptions | PreferredGroup, preferredGroup?: PreferredGroup): Promise<IEditorPane | undefined> {
 		let typedEditor: IEditorInput | undefined = undefined;
 		let options = isEditorInput(editor) ? optionsOrPreferredGroup as IEditorOptions : editor.options;
 		let group: IEditorGroup | undefined = undefined;
 		let activation: EditorActivation | undefined = undefined;
 
-		if (isOpenInEditorGroup(optionsOrPreferredGroup)) {
+		if (isPreferredGroup(optionsOrPreferredGroup)) {
 			preferredGroup = optionsOrPreferredGroup;
 		}
 
 		// Resolve override unless disabled
 		if (options?.override !== EditorOverride.DISABLED) {
-			const [resolvedEditor, resolvedGroup, resolvedActivation] = await this.doResolveEditor(isEditorInput(editor) ? { editor, options } : editor, preferredGroup);
+			const resolvedEditor = await this.editorOverrideService.resolveEditorInput(isEditorInput(editor) ? { editor, options } : editor, preferredGroup);
 
 			if (resolvedEditor === OverrideStatus.ABORT) {
 				return; // skip editor if override is aborted
 			}
 
-			group = resolvedGroup;
-			activation = resolvedActivation;
```

Comment:
We lost `activation` now, it does not seem to be set anymore?

Reply:
Activation gets set in the service.

Reply:
Hm I see. Can we maybe make this more obvious by moving the code around this comment:

```
// Mixin editor group activation if any
```

Into where we call `findGroup` above? It seems that is the only place where we would resolve a `activation` outside the editor override service.

Reply:
I will push that.